packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarousel/useTeachingPopoverCarousel.ts

```diff
@@ -3,115 +3,136 @@ import {
+import { useSyncExternalStore } from 'use-sync-external-store/shim';
 import { usePopoverContext_unstable } from '@fluentui/react-popover';
 import { TeachingPopoverCarouselNav } from '../TeachingPopoverCarouselNav/TeachingPopoverCarouselNav';
+import { useCarousel_unstable } from './Carousel/Carousel';
 
 export const useTeachingPopoverCarousel_unstable = (
   props: TeachingPopoverCarouselProps,
   ref: React.Ref<HTMLDivElement>,
 ): TeachingPopoverCarouselState => {
-  const { layout = 'centered', paginationType = 'icon', strings, onPageChange, onFinish } = props;
+  const {
+    layout = 'centered',
+    paginationType = 'icon',
+    onPageChange,
+    onFinish,
+    currentPage,
+    defaultCurrentPage,
+  } = props;
+  const { carousel, carouselWalker, carouselRef } = useCarousel_unstable();
+
+  const { store, setValue, setIndex, currentIndex: _currentPage, totalPages: _totalPages, value } = carousel;
 
   const appearance = usePopoverContext_unstable(context => context.appearance);
-  const triggerRef = usePopoverContext_unstable(context => context.triggerRef);
   const toggleOpen = usePopoverContext_unstable(context => context.toggleOpen);
 
-  const reactChildArray = React.Children.toArray(props.children);
+  const values = useSyncExternalStore(store.subscribe, () => store.getSnapshot());
+  const totalPages = values.length;
 
-  const [totalPages, setTotalPages] = useState(reactChildArray.length);
-
-  // ToDo: Imperative setCurrentPage hook
-  const [currentPage, setCurrentPage] = useControllableState({
-    initialState: 0,
-    defaultState: props.defaultCurrentPage,
-    state: props.currentPage,
-  });
+  useIsomorphicLayoutEffect(() => {
+    if (value && !values.includes(value)) {
+      // Safety check that the page we are on hasn't been removed
+      // Stick to current index so we won't impact UI if things change
+      setIndex(_currentPage);
+    }
+  }, [values, value, _currentPage, setIndex]);
 
-  React.useEffect(() => {
-    // Update total pages if child length changes.
-    if (totalPages !== reactChildArray.length) {
-      setTotalPages(reactChildArray.length);
+  useIsomorphicLayoutEffect(() => {
+    // Handles default initialization page on mount
+    if (defaultCurrentPage !== undefined && defaultCurrentPage !== _currentPage) {
+      setIndex(defaultCurrentPage);
     }
-  }, [reactChildArray.length, totalPages, setTotalPages]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
 
-  // Get current pagination child (w/ array safety)
-  const currentPageElement =
-    currentPage < reactChildArray.length ? reactChildArray[currentPage] : reactChildArray[reactChildArray.length - 1];
+  useIsomorphicLayoutEffect(() => {
+    // Handles external updates of currentPage via props
+    if (currentPage !== undefined && currentPage !== _currentPage) {
+      setIndex(currentPage);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [currentPage]);
```

Comment:
This is no-go: we should avoid syncing states, that's not the goal. Instead `useCarousel_unstable()` should control state based on users input